.github/workflows/ci-golang-lint.yml

```diff
@@ -0,0 +1,43 @@
+name: ci-golang-lint
+on:
+  push:
+    branches:
+      - main
+      - release/*
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches:
+      - main
+      - release/*
+  schedule:
+    - cron: "0 * * * *"
+jobs:
+  golang-lint:
+    if: github.event.pull_request.draft == false
+    name: lint
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get Go Version
+        run: |
+          #!/bin/bash
+          GOVERSION=$({ [ -f .go-version ] && cat .go-version; })
+          echo "GOVERSION=$GOVERSION" >> $GITHUB_ENV
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: golang-lint
+        env:
+          # prevent OOM
+          GOGC: 10
+        uses: golangci/golangci-lint-action@v3
```

Comment:
I vaguely remember that go-sec changes may need to be brought in for this to pass. I would suggest to not run this if this is going to fail. You can also control the linters from the linter-dot-file in the project directory.

Reply:
@skrdgraph That is what I thought but it looks like it is passing.  I double checked the .golanglint config file and we are only running the `lll` linter.  I am not sure when we commented `gas` because Matthew made some fixes for that (I pulled the lint config file from main).

Reply:
yeah - long story short, we did comment out `gas` because of some specific new failures. I will create a new issue for this - and look into this.